server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts

```diff
@@ -0,0 +1,251 @@
+import TypeOfWorkExperienceValue from '../../../enums/typeOfWorkExperienceValue'
+import toCreateOrUpdateInductionDto from '../../../data/mappers/createOrUpdateInductionDtoMapper'
+import previousWorkExperienceTypeScreenOrderComparator from '../previousWorkExperienceTypeScreenOrderComparator'
+
+export default class PreviousWorkExperienceTypesUpdateController extends PreviousWorkExperienceTypesController {
+  constructor(private readonly inductionService: InductionService) {
+    super()
+  }
+
+  getBackLinkUrl(req: Request): string {
+    const { prisonNumber } = req.params
+    return `/plan/${prisonNumber}/view/work-and-interests`
+  }
+
+  getBackLinkAriaText(_req: Request): string {
+    return 'Back to <TODO - check what CIAG UI does here>'
+  }
+
+  submitPreviousWorkExperienceTypesForm: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonNumber } = req.params
+    const { prisonerSummary, inductionDto } = req.session
+
+    req.session.previousWorkExperienceTypesForm = { ...req.body }
+    if (!req.session.previousWorkExperienceTypesForm.typeOfWorkExperience) {
+      req.session.previousWorkExperienceTypesForm.typeOfWorkExperience = []
+    }
+    if (!Array.isArray(req.session.previousWorkExperienceTypesForm.typeOfWorkExperience)) {
+      req.session.previousWorkExperienceTypesForm.typeOfWorkExperience = [
+        req.session.previousWorkExperienceTypesForm.typeOfWorkExperience,
+      ]
+    }
+    const { previousWorkExperienceTypesForm } = req.session
+
+    const errors = validatePreviousWorkExperienceTypesForm(previousWorkExperienceTypesForm, prisonerSummary)
+    if (errors.length > 0) {
+      req.flash('errors', errors)
+      return res.redirect(`/prisoners/${prisonNumber}/induction/previous-work-experience`)
+    }
+
+    if (
+      noChangesToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm) ||
+      onlyRemovalsWereMadeToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm)
+    ) {
+      if (onlyRemovalsWereMadeToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm)) {
+        logger.debug(
+          'Previous Work Experiences changes are only removed types so Induction can be updated without starting a flow',
+        )
+        // call the service to update the Induction
+        try {
+          await this.updateInduction(inductionDto, previousWorkExperienceTypesForm, req, prisonerSummary)
+        } catch (e) {
+          return next(createError(500, `Error updating Induction for prisoner ${prisonNumber}. Error: ${e}`))
+        }
+      } else {
+        logger.debug('No changes to Previous Work Experiences were submitted')
+      }
+
+      req.session.previousWorkExperienceTypesForm = undefined
+      req.session.inductionDto = undefined
+      return res.redirect(`/plan/${prisonNumber}/view/work-and-interests`)
+    }
```

Comment:
The PR adds the unit test coverage for everything up until here (including this `if` block for "no changes" and "only removals")

The following section (which is not complete) is not covered, but will be tested in my next PR when we introduce the "mini-flow"